src/buyflow/Buyflow.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import AgeStep from './AgeStep'
+import EmailStep from './EmailStep'
+import SummaryStep from './SummaryStep'
+
+interface BuyflowProps {
+  productId: ProductIds
+}
+
+export enum ProductIds {
+  devIns = 'dev_ins',
+}
+
+const PRODUCT_IDS_TO_NAMES = {
+  [ProductIds.devIns]: 'Developer Insurance',
+}
+
+const Buyflow: React.FC<BuyflowProps> = (props) => {
+  const [currentStep, setStep] = useState('email')
+  const [collectedData, updateData] = useState({
+    email: '',
+    age: 0,
+  })
```

Comment:
depending on the product resulted data can have a different shape, so we won't assume it
```suggestion
  const [collectedData, updateData] = useState({})
```